test/validation/api/ipsec/ipsec_test_out.c

```diff
@@ -446,6 +482,16 @@ static void test_out_in_common(ipsec_test_flags *flags,
 		},
 	};
 
+	if (flags->v6 && !flags->tunnel) {
```

Comment:
The v6 flag has a problematic double meaning. In tunnel mode it tells whether the tunnel is IPv6 or IPv4 and in transport mode it tells if the payload packet is IPv6 or IPv4. Now if I want to add a test for tunneled IPv6 packet, the logic related to the flag breaks apart. It would be better to just define two different flags for the two different things already now. And once you do that, you could even add a tests for IPv6 payloads in UDP encapsulated tunnels.

Reply:
So I'll update flags as,

```
odp_bool_t v6;
odp_bool_t tunnel;
odp_bool_t v6_tunnel;
```

The first one will say test packet's IP version. 

Or should I go back to having v4_tunnel & v6_tunnel? If both are not set, then that would indicate transport. What is your preference?

Reply:
Either way is ok for me. Maybe the v6_tunnel in the example above could be something like tunnel_is_v6 to make it clearer that tunnel is not just a shorter version of v4_tunnel but an overall tunneling flag.

Reply:
I tried enabling all combinations,

```
  Test: test_udp_encap ...
        IPv4 Transport
        IPv6 Transport
        IPv4-in-IPv4 Tunnel
        IPv4-in-IPv6 Tunnel
        IPv6-in-IPv4 Tunnel
        IPv6-in-IPv6 Tunnel
  passed
```